File: 351003/Matelenok/task1/src/main/java/by/distcomp/task1/mapper/UserMapper.java
package by.distcomp.task1.mapper;

import by.distcomp.task1.dto.UserRequestTo;
import by.distcomp.task1.dto.UserResponseTo;
import by.distcomp.task1.model.User;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface UserMapper {
    User toEntity(UserRequestTo dto);

    UserResponseTo toResponse(User user);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/UserMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.user.UserRequestTo
import com.distcomp.dto.user.UserResponseTo
import com.distcomp.entity.User
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface UserMapper {
    fun toUserResponse(user: User) : UserResponseTo

    fun toUserEntity(userRequestTo: UserRequestTo) : User
}